.github/actions/repo-state/action.yml

```diff
@@ -0,0 +1,51 @@
+name: 'Repository state'
+description: 'Get the latest tag, the latest process-note, the number of commits since the last tag and the version numbers from the last tag.'
+outputs:
+  tag:
+    description: "Last tagged version"
+    value: ${{ steps.tag.outputs.value }}
+  note:
+    description: "Note on the last tag"
+    value: ${{ steps.note.outputs.value }}
+  commits:
+    description: "Number of commits since the last tag"
+    value: ${{ steps.commits.outputs.value }}
+  version_major:
+    description: "Major version number"
+    value: ${{ steps.version.outputs.major }}
+  version_minor:
+    description: "Minor version number"
+    value: ${{ steps.version.outputs.minor }}
+  version_revision:
+    description: "Revision number"
+    value: ${{ steps.version.outputs.revision }}
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: 'master'
+    - name: Get the last release-tag
+      id: tag
+      shell: bash
+      run: echo "value=$(git describe --tags --abbrev=0)" >> $GITHUB_OUTPUT
+    - name: Get the note on the last release-tag
+      id: note
+      shell: bash
+      run: |
+        git fetch origin refs/notes/*:refs/notes/*
+        echo 'value<<EOF' >> $GITHUB_OUTPUT
+        (git notes show $(git describe --tags --abbrev=0) | cat || true) >> $GITHUB_OUTPUT
```

Comment:
Notes are the only way to add information to a commit in the past without altering its commit-hash.

I agree that notes are horribly implemented in most libraries/clients, that is why I would never use them for actual development, but it turns out they are very useful for CI/buildtools. The worst thing that can happen here is that we release commits without actual changes to the binaries.

The only other way I know to solve this is to have a second repository that is only used to store build-information.
So you would have `project/project-repo` and `project/buildinfo`. But this usually only used if there is a lot to store e.g. if someone would like to keep information about every build that ever happens.

We never need to read or edit the notes by hand so it does not matter how they are handled as log as the workflow does it correctly.

Reply:
What about storing some information inside the github release page? I'm not saying that it is a better option yet, but it is an alternative, probably with its own drawbacks to weigh against using notes.

Reply:
I would rather not show the sha256 hashes to the user, because they are not reproducible with the binaries they find in the release section (at the moment this only true for `burrito_link.exe`).

While git notes are not well supported by many clients, they are a stable feature for over 10 years.
If there is no scenario that clearly breaks the implementation in the workflow I see this as the best solution to the problem.

Reply:
Sound good. I agree.